Core/include/Acts/Visualization/IVisualization.hpp

```diff
@@ -65,9 +65,11 @@ class IVisualization {
                     ColorType color = {120, 120, 120}) = 0;
 
   /// Write the content of the helper to an outstream.
-  /// @param os The output stream
+  /// @param os The output stream for file
+  /// @param ose The (optional pointer) to an extra stream
+  /// e.g. for material definitions
   ///
-  virtual void write(std::ostream& os) const = 0;
+  virtual void write(std::ostream& os, std::ostream* eos = nullptr) const = 0;
```

Comment:
Ok, so we talked about this this morning. If I look at it like this, I'm not sure it needs to be in the interface. Look at it this way: to use this parameter correctly, you have to know what kind of helper you're dealing with anyway, at the very least to give the extra file a proper name. If you write generic code that accepts any helper and treats them the same, you end up with an empty file "extra.mtl" that the PLY writer didn't use.

The alternative idea would be to just add an additional overload in the `ObjHelper` with this signature, that will call the overriding function matching the virtual function signature here, and additionally handle the extra stream. That way it's completely explicit.

What do you think?

Reply:
That's how I started out, *however* I ran in into a problem: `fstream` or any `ostream` does not allow to trace back the file name, otherwise I had just opened a new stream in the `write()` method.

If I write and `Extra.mtl` file and I have a bunch of unit tests executed in one directory, then - unfortunately - this one is overwritten and the output non-sensical.

The way I have implemented now is that the `write()` method does not know anything about file naming, in fact it doesn't even know it's an `fstream` (which I would implicitly assume if I opened the file within the write() method.

Reply:
As discussed in the daily meeting, let's change this to two overloads:

```cpp
virtual IVisualization::write(std::ostream& os);
// EITHER
virtual IVisualization::write(std::fstream& fs);
// OR
virtual IVisualization::write(std::filesystem::path& path);
```
and no other overloads here.

And then the `ObjVisualization` can do

```cpp
std::ostream& ObjVisualization::write(std::ostream& os) override; // this is the basic override
std::ostream& ObjVisualization::write(std::ostream& os, std::ostream& mtl); // not virtual, ONLY on OBJ
std::ostream& ObjVisualization::write(std::filesystem::path& path) { // not virtual, not an override, ONLY provided in OBJ

   std::fstream fs{path};
   std::filesystem::path mtlPath = deriveMaterialFileName(path;
   std::fstream mtl{mtlPath};

   return write(fs, mtl);
}
```

Reply:
Agreed, I will change accordingly.